plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/GitServiceClient.java

```diff
@@ -68,6 +68,31 @@
    */
   Promise<Void> fetch(Path project, String remote, List<String> refspec, boolean removeDeletedRefs);
 
+  /**
+   * Fetch changes from remote repository to local one (sends request over WebSocket).
+   *
+   * @param project project root of GIT repository
+   * @param remote remote repository's name
+   * @param refspec list of refspec to fetch.
+   *     <p>Expected form is:
+   *     <ul>
+   *       <li>refs/heads/featured:refs/remotes/origin/featured - branch 'featured' from remote
+   *           repository will be fetched to 'refs/remotes/origin/featured'.
+   *       <li>featured - remote branch name.
+   *     </ul>
+   *
+   * @param removeDeletedRefs if <code>true</code> then delete removed refs from local repository
+   * @param username username to perform vcs authorization
+   * @param password password to perform vcs authorization
+   */
+  Promise<Void> fetch(
+      Path project,
+      String remote,
+      List<String> refspec,
+      boolean removeDeletedRefs,
+      String username,
```

Comment:
I think it is better to use `org.eclipse.che.ide.ui.dialogs.askcredentials.Credentials`  to pass this object to parameters instead of Strings.

Reply:
I'm not sure that class from `.ui.` package is well suitable here.

Reply:
May be moving this class to some more suitable location will solve this problem. BTW this class is not used, so it will be easy to do that.